coverage/jsonreport.py

```diff
@@ -122,6 +126,45 @@ def report_one_file(self, coverage_data: CoverageData, analysis: Analysis) -> di
             reported_file["missing_branches"] = list(
                 _convert_branch_arcs(analysis.missing_branch_arcs()),
             )
+
+        for region in file_reporter.code_regions():
+            if region.kind not in reported_file:
+                reported_file[region.kind] = {}
+            num_lines = len(file_reporter.source().splitlines())
+            outside_lines = set(range(1, num_lines + 1))
+            outside_lines -= region.lines
+            narrowed_analysis = analysis.narrow(region.lines)
+            narrowed_nums = narrowed_analysis.numbers
+            narrowed_summary = {
+                "covered_lines": narrowed_nums.n_executed,
+                "num_statements": narrowed_nums.n_statements,
+                "percent_covered": narrowed_nums.pc_covered,
+                "percent_covered_display": narrowed_nums.pc_covered_str,
+                "missing_lines": narrowed_nums.n_missing,
+                "excluded_lines": narrowed_nums.n_excluded,
+            }
+            reported_file[region.kind][region.name] = {
+                "executed_lines": sorted(narrowed_analysis.executed),
+                "summary": narrowed_summary,
+                "missing_lines": sorted(narrowed_analysis.missing),
+                "excluded_lines": sorted(narrowed_analysis.excluded),
+            }
+            if self.config.json_show_contexts:
+                contexts = coverage_data.contexts_by_lineno(narrowed_analysis.filename)
+                reported_file[region.kind][region.name]["contexts"] = contexts
+            if coverage_data.has_arcs():
+                narrowed_summary.update({
+                    "num_branches": narrowed_nums.n_branches,
+                    "num_partial_branches": narrowed_nums.n_partial_branches,
+                    "covered_branches": narrowed_nums.n_executed_branches,
+                    "missing_branches": narrowed_nums.n_missing_branches,
```

Comment:
When I change this to `"xmissing_branches"`, no tests fail, so it seems like we're missing a test of regions with branches?

Reply:
I can take over finishing this.

Reply:
Feel free to, I probably won't have time to work on it today and would love to see it land.